src/storage-client/src/controller/hosts.rs

```diff
@@ -74,7 +74,7 @@ pub struct StorageHosts<T> {
     /// The known storage hosts, identified by network address.
     hosts: HashMap<StorageHostAddr, StorageHost<T>>,
     /// The assignment of storage objects to storage hosts.
-    objects: HashMap<GlobalId, StorageHostAddr>,
+    objects: Arc<Mutex<HashMap<GlobalId, StorageHostAddr>>>,
```

Comment:
I *think* this doesnt produce a reference cycle, but we have to be careful!

Reply:
Also, this should not be a tokio mutex, it should be a std mutex, as its protecting plain data: https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use

Reply:
done!